pwdtk/middlewares.py

```diff
@@ -38,58 +33,26 @@ def __init__(self, get_response=None):
         else:
             super(PwdtkMiddleware, self).__init__()
 
-    def must_renew_password(self, request):
-
-        if request.method == 'GET':
-            return False
-
-        if django.VERSION < (1, 10):
-            is_authenticated = request.user and request.user.is_authenticated()
-        else:
-            is_authenticated = request.user and request.user.is_authenticated
-
-        if not is_authenticated:
-            return False
-
-        if not PwdData.get_or_create_for_user(request.user).must_renew:
-            return False
-
-        if request.path == reverse(PwdtkSettings.PWDTK_PASSWD_CHANGE_VIEW):
-            return False
-
-        if request.path in PwdtkSettings.PWDTK_PASSWD_CHANGE_ALLOWED_PATHS:
-            return False
-
-        return True
+    def process_request(self, request):
+        if hasattr(PwdtkSettings, "reset_cache"):
+            PwdtkSettings.reset_cache()
 
     def process_exception(self, request, exception):
 
         if isinstance(exception, PwdtkLockedException):
             context = exception.pwdtk_data.get_lockout_context()
-            if request.is_ajax():
-                return HttpResponse(
-                    json.dumps(context),
-                    content_type='application/json',
-                    status=403,
-                    )
-
-            if PwdtkSettings.PWDTK_LOCKOUT_VIEW:
-                return redirect("%s?%s" % (
-                    reverse(PwdtkSettings.PWDTK_LOCKOUT_VIEW),
-                    urlencode(context)
-                    )
+            context["status"] = "PWDTK_LOCKED"
+            return HttpResponse(
+                json.dumps(context),
+                content_type='application/json',
+                status=403,
                 )
-            return lockout_response(request, exception.pwdtk_data)
-        return None
-
-    def process_request(self, request):
 
-        if self.must_renew_password(request):
-            return redirect(reverse(PwdtkSettings.PWDTK_PASSWD_CHANGE_VIEW))
-
-    def process_response(self, request, response):
-
-        if self.must_renew_password(request):
-            return redirect(reverse(PwdtkSettings.PWDTK_PASSWD_CHANGE_VIEW))
+        if isinstance(exception, PwdtkForceRenewException):
+            return HttpResponse(
+                json.dumps({"status": "PWDTK_NEED_RENEW_PASSWORD"}),
```

Comment:
not sure I understand what this means.
PWDTK_NEED_RENEW_PASSWORD


Does this mean we must renew the password to authenticate?

`PWDTK_MUST_RENEW_PASSWORD`

or

`PWDTK_NEEDS_RENEW_PASSWORD` (NEEDSA with an S)

Reply:
it is more like NEEDS, wrong typo but maybe we can keep this

Reply:
PWDTK_NEED_RENEW_PASSWORD status is already used